### ISSUE TITLE
IE11 and Edge error

### ISSUE THREAD

**Issue by thul** [cksource/ckeditor-plugin-a11ychecker#219]
I have a project where i use Ckeditor with the a11ychecker, i use the following versions: CKeditor 4.5.9, A11ychecker 1.0.0, Balloonpanel 4.5.9 with jQuery 3.x.

When clicking the a11ychecker button it causes a javascript error "_Object doesn't support property or method 'exec'_ ". Which occurs in here https://github.com/cksource/ckeditor-plugin-a11ychecker/blob/5d497a5ade75e9fbe46c4bb70880197ede1c4152/plugin.js#L224

It works fine in Safari, Chrome and Firefox. It's just IE11 and Edge that are giving problems.

This is the config im using

```javascript
{
                language: 'nl',
                skin: 'icy_orange',
                forceCustomDomain: true,
                width: '100% -2px',
                toolbarLocation: 'top',
                format_tags: 'p;h2;h3;h4;h5;h6',
                extraPlugins: 'showblocks,a11ychecker,balloonpanel',
                a11ychecker_quailParams: {
                    guideline: [
                        'aAdjacentWithSameResourceShouldBeCombined',
                        'aImgAltNotRepetitive',
                        'aLinksAreSeparatedByPrintableCharacters',
                        'aMustNotHaveJavascriptHref',
                        'aSuspiciousLinkText',
                        'documentVisualListsAreMarkedUp',
                        'headerH1',
                        'headerH2',
                        'headerH3',
                        'imgAltIsDifferent',
                        'imgAltIsTooLong',
                        'imgAltNotEmptyInAnchor',
                        'imgAltTextNotRedundant',
                        'imgHasAlt',
                        'linkHasAUniqueContext',
                        'pNotUsedAsHeader',
                        'tableDataShouldHaveTh',
                        'tableNotUsedForLayout'
                    ]
                },
                toolbar: 'Basic',
                toolbar_Basic: [
                    { name: 'tools', items: [ 'A11ychecker' ] },
                    { name: 'document', items: [ 'Source' ] },
                    { name: 'styles', items: [ 'Format' ] },
                    { name: 'basicstyles', items: [ 'Bold', 'Italic' ] },
                    { name: 'lists', items: ['NumberedList', 'BulletedList']},
                    { name: 'tools', items: [ 'ShowBlocks' ] },
                    { name: 'links', items: [ 'Link', 'Unlink'] },
                    { name: 'images', items: [ 'Image'] },
                    { name: 'clipboard', items: [ 'PasteText' ] }
                ]
            };
```

Am i doing something wrong? is it a bug? what would solve this issue im having? Any help / suggestions are welcome

**Comment by mlewand** [comment]
Is it happening only for built or dev version? Or is it happening for both Accessibility Checker versions?

**Comment by thul** [comment]
A built package from the dev version or a checkout of the code both result into an error "require is not defined" when loading CKeditor. In all browsers this time, i followed the README to the letter for this too.

**Comment by mlewand** [comment]
You're right, because dev version has a dependency on RequireJS. I've updated the readme so that it mentions this. However RequireJS needs also be configured correctly to discover the modules, see #220.

OK so it's my understanding that you've been experiencing these problems on a built, release version distributed in our add-ons repository http://ckeditor.com/addon/a11ychecker.

We'll check it with IE11/Edge and update the ticket as soon as we're able to reproduce the issue.

**Comment by thul** [comment]
For what it's worth, we load our CKeditor using an Angular directive. But that shouldn't affect plugins as far as i know.

**Comment by Breuls** [comment]
I've taken some time to look into this problem, not knowing anything about the internals of CKEditor or the a11ychecker plugin. There are a couple of things I noticed when testing this in Edge:

There are some problems during load time of the editor and the plugin. One is that a 'beforeInit' doesn't complete past [this line](https://github.com/cksource/ckeditor-plugin-a11ychecker/blob/master/plugin.js#L69):

    var a11ychecker = new Controller( editor ),

I have done some debugging (using console.log and console.trace calls), and code past this line does not seem to be executed. I debugged this in the minified version, by the way, so I actually did this:

	try {
	    var i = new n(t);
	    console.log('works')
	} catch (e) {
	    console.log('error');
	    console.log(e);
	}

The `new n` is the minified version of `new Controller`. The 'works' line is never printed, but the 'error' line is. The exception object 'e' shows that a `TypeError` with the message 'Permission denied' occurred.

Similarly, this error occurs during another part of the load process, but this time inside CKEditor code. [This line](https://github.com/ckeditor/ckeditor-dev/blob/12f0de314fd6fbee0bc4d35d541123d283fdecc9/core/dom/domobject.js#L212) throws that same error:

    var expandoNumber = this.$[ 'data-cke-expando' ],

The issue here is that `$` does not have an accessible value, where in other situations, it's either the document body or an object. A console.trace reveals that this line is reached from [EditableDecorator.prototype.addListeners](https://github.com/cksource/ckeditor-plugin-a11ychecker/blob/master/EditableDecorator.js#L112), as evidenced by this trace output:

	console.trace()
	   at a.getCustomData (http://local.dev/js/ckeditor/ckeditor.js?dev:1135:17)
	   at on (http://local.dev/js/ckeditor/ckeditor.js?dev:1095:17)
	   at proto.on (http://local.dev/js/ckeditor/ckeditor.js?dev:5708:21)
	   at proto.attachListener (http://local.dev/js/ckeditor/ckeditor.js?dev:5712:71)
	   at t.prototype.addListeners (http://local.dev/js/ckeditor/plugins/a11ychecker/plugin.js?t=G6DD:287:13)
	   at t (http://local.dev/js/ckeditor/plugins/a11ychecker/plugin.js?t=G6DD:257:13)
	   at u (http://local.dev/js/ckeditor/plugins/a11ychecker/plugin.js?t=G6DD:890:13)
	   at Anonymous function (http://local.dev/js/ckeditor/plugins/a11ychecker/plugin.js?t=G6DD:1072:29)
	   at Anonymous function (http://local.dev/js/ckeditor/plugins/a11ychecker/plugin.js?t=G6DD:1067:21)
	   at a[1] (http://local.dev/js/ckeditor/ckeditor.js?dev:137:25)

The line numbers might not be useful (they are the result of formatting minified files), but the function and file names might give some insight into the context.

These load errors occur in Edge, and not in a browser where the a11ychecker button works, which leads to the conclusion that the one situation might cause the other. 

Going back to that button; I've looked into the context in an Edge and a Chrome environment. Did some console.logging, and in the `generateCommand()` call mentioned in the issue, it seems that the error is thrown because the `controllerMethod` ('exec') isn't available in the prototype. For context, it's this line:

    editor._.a11ychecker[ controllerMethod ]

Reading the code leads to the assumption that there must be an element 'exec' inside `editor._.a11ychecker` for this to work. However, this isn't the case in Chrome, which does not throw an error. However, when looking at the `__proto__` accessor, it shows that that does have an exec function. In Edge, looking at the same variables through console.log inspection, shows that the `__proto__` is simply empty, as is the `a11ychecker` property itself.

For reference and context, the output from Edge:

	[object Object]
	plugin.js (1027,13)
	   {
	      [functions]: ,
	      _: {
	         [functions]: ,
	         __proto__: {
	            [functions]: ,
	            __proto__: null
	         },
	         a11ychecker: {
	            [functions]: ,
	            __proto__: {
	               [functions]: ,
	               __proto__: null
	            }
	         },
	         (... shortened ...)

Chrome doesn't allow for a indented object representation to be copied, but suffice to say that the `a11ychecker` property and its prototype are nicely filled with methods and other properties.

I can't take this any further, as I'm limited by my knowledge of javascript, CKEditor and a11ychecker, but I hope I have provided some context that can be of help with investigating this issue.
